### ISSUE TITLE
Start shuffling questions after 100, not 250

### ISSUE THREAD

**Issue by duogenesis** [duolicious/duolicious-backend#24]
The question bank has been ordered so that questions of better quality some first. Consequently, users would have the best experience if they answer them in-order, given the current matching algorithm.

But I'd like to improve the algorithm in the future. I plan to do that with an auto encoder. But I need training data for that. If only a very small minority of users ever answer the later questions, I wouldn't have enough training data to predict answers to those later questions, given users' initial answers. So I'm currently randomising the order of the questions, but only after users have answered 250 questions. Having the order of the initial questions be fixed and the later questions be random strikes the balance between (1) optimising the performance of the current algorithm, and (2) optimising the size of the training set I can use to build a better model.

The "250" threshold was chosen by feel, after I tested the personality assessment on myself, to check how few answers I needed to give before the system accurately determined my personality. But while 250 questions was better, 100 worked well too. I've also gotten more data since my initial self-test. I had 50 other users sign up by posting the app on a social media website. Based on their usage, a threshold of 100 would increase the percentage of users who get past the initial, fixed portion of the question bank from about 3% to about 35%:

```
postgres=# select row_number() over (order by count_answers desc), percent_rank() over (order by count_answers desc), count_answers from person;
 row_number | percent_rank | count_answers 
------------+--------------+---------------
          1 |            0 |          1200
          2 |         0.02 |           564 <- Only 2 out of 51 people answered 250 questions or more
          3 |         0.04 |           242    (and I suspect the person who answered 1200 answered randomly)
          4 |         0.06 |           241
          5 |         0.08 |           226
          6 |          0.1 |           217
          7 |         0.12 |           203
          8 |         0.14 |           186
          9 |         0.16 |           183
         10 |         0.18 |           158
         11 |          0.2 |           148
         12 |         0.22 |           132
         13 |         0.24 |           115
         14 |         0.26 |           114
         15 |         0.28 |           113
         16 |          0.3 |           111
         17 |         0.32 |           110
         18 |         0.34 |           105  <- 18 out of 51 people answered at least 100 questions
         19 |         0.36 |            90
         20 |         0.38 |            87
         21 |          0.4 |            84
         22 |         0.42 |            83
         23 |         0.44 |            72
         24 |         0.46 |            62
         25 |         0.48 |            53
         26 |          0.5 |            50
         27 |         0.52 |            45
         28 |         0.54 |            42
         29 |         0.56 |            41
         30 |         0.56 |            41
         31 |          0.6 |            40
         32 |         0.62 |            35
         33 |         0.64 |            30
         34 |         0.66 |            25
         35 |         0.68 |            24
         36 |         0.68 |            24
         37 |         0.72 |            21
         38 |         0.74 |            16
         39 |         0.76 |            14
         40 |         0.76 |            14
         41 |          0.8 |            12
         42 |         0.82 |             9
         43 |         0.84 |             8
         44 |         0.86 |             5
         45 |         0.88 |             3
         46 |         0.88 |             3
         47 |         0.92 |             1
         48 |         0.94 |             0
         49 |         0.94 |             0
         50 |         0.94 |             0
         51 |         0.94 |             0
```

**Comment by duogenesis** [comment]
nvm, this is already done.
